### ISSUE TITLE
Cop idea: Referencing named capture with a number

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#7754]
## Is your feature request related to a problem? Please describe.

Regexp's named capture can be referenced with a number, but referencing with a name is preferred.

```ruby
# bad
'str'.match(/(?<X>s)/)[1]
# => "s"

# good
'str'.match(/(?<X>s)/)[:X]
# => "s"

# also good
'str'.match(/(?<X>s)/)['X']
# => "s"
``` 

## Describe the solution you'd like

Add a cop to detect the problem.

## Describe alternatives you've considered

Nothing

## Additional context

We will be able to the cop with regexp_parser gem. ref: #7746

**Comment by choosen** [comment]
should we detect also following example? :
```
match_data = 'str'.match /(?<X>s)/
match_data[1]
```

**Comment by pocke** [comment]
I think the cop should detect the problem ideally, but I guess it is difficult with the current RuboCop implementation.Because RuboCop's analysis for local variable assigments is not enougth.
I think a simple implementation, which detects no-assignment cases only, is enough for the first step.
